govcd/lbappprofile.go

```diff
@@ -0,0 +1,204 @@
+					", but its ID (%s) does not match specified ID (%s)",
+					profile.Name, profile.ID, lbAppProfileConfig.ID)
+			}
+			return profile, nil
+		}
+	}
+
+	return nil, fmt.Errorf("could not find load balancer application profile (name: %s, ID: %s)",
+		lbAppProfileConfig.Name, lbAppProfileConfig.ID)
+}
+
+// ReadLBAppProfileByID wraps ReadLBAppProfile and needs only an ID for lookup
+func (eGW *EdgeGateway) ReadLBAppProfileByID(id string) (*types.LBAppProfile, error) {
+	return eGW.ReadLBAppProfile(&types.LBAppProfile{ID: id})
+}
+
+// ReadLBAppProfileByName wraps ReadLBAppProfile and needs only a Name for lookup
+func (eGW *EdgeGateway) ReadLBAppProfileByName(name string) (*types.LBAppProfile, error) {
+	return eGW.ReadLBAppProfile(&types.LBAppProfile{Name: name})
+}
+
+// UpdateLBAppProfile updates types.LBAppProfile with all fields. At least name or ID must be specified.
+// If both - Name and ID are specified it performs a lookup by ID and returns an error if the specified name and found
+// name do not match.
+func (eGW *EdgeGateway) UpdateLBAppProfile(lbAppProfileConfig *types.LBAppProfile) (*types.LBAppProfile, error) {
+	if err := validateUpdateLBAppProfile(lbAppProfileConfig); err != nil {
+		return nil, err
+	}
+
+	// if only name was specified for update, ID must be found, because ID is mandatory for update
+	if lbAppProfileConfig.ID == "" {
+		readlbAppProfile, err := eGW.ReadLBAppProfile(&types.LBAppProfile{Name: lbAppProfileConfig.Name})
+		if err != nil {
+			return nil, fmt.Errorf("unable to find load balancer application profile by name for update: %s", err)
+		}
+		lbAppProfileConfig.ID = readlbAppProfile.ID
+	}
+
+	httpPath, err := eGW.buildProxiedEdgeEndpointURL(types.LBAppProfilePath + lbAppProfileConfig.ID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+
+	// Result should be 204, if not we expect an error of type types.NSXError
+	_, err = eGW.client.ExecuteRequestWithCustomError(httpPath, http.MethodPut, types.AnyXMLMime,
+		"error while updating load balancer application profile : %s", lbAppProfileConfig, &types.NSXError{})
+	if err != nil {
+		return nil, err
+	}
+
+	readAppProfile, err := eGW.ReadLBAppProfile(&types.LBAppProfile{ID: lbAppProfileConfig.ID})
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve application profile with ID (%s) after update: %s",
+			readAppProfile.ID, err)
+	}
+	return readAppProfile, nil
+}
+
+// DeleteLBAppProfile is able to delete the types.LBAppProfile type by Name and/or ID.
+// If both - Name and ID are specified it performs a lookup by ID and returns an error if the specified name and found
+// name do not match.
+func (eGW *EdgeGateway) DeleteLBAppProfile(lbAppProfileConfig *types.LBAppProfile) error {
+	if err := validateDeleteLBAppProfile(lbAppProfileConfig); err != nil {
+		return err
+	}
+
+	lbAppProfileID := lbAppProfileConfig.ID
```

Comment:
Seems getting code part duplicates through methods and could be moved to separate method

Reply:
I split it. How does it look now. If it's fine I will retrofit the other components.

Reply:
I think better :)

Reply:
Made the same thing for other components.

Reply:
nice